### PR TITLE
New login design

### PR DESCRIPTION
**Pull request by StylianosGakis**
Since this grew up in size a bit, since I am trying to get it to work asap and didn't have the time to split things up as much as I'd like, I think it'd help you review this by mainly looking at the commit messages. Those definitely tell a better story of how all this came to be. I am trying to do a good job explaining what each one changes, while the big PR itself might be harder to decipher from just looking at the code changes.

So one thing that may be a bit confusing to you here (rightly so) is why there is some code changing some feature flag from HAnalytics things, and why I removed us sending HAnalytics events here. So as I was adding some code, we had a circular dependency issue with Koin (nothing to do with Koin, it was a legit circular reference issue) which looked something like this:

To make OkHttp instance, we had some interceptor, `AuthTokenRefreshingInterceptor` which needed to have access to `AccessTokenProvider` to get the token, which in turn needed the one below and so on, until it needed OkHttp again. So it was failing. The fix was to try and remove some of the code which I knew we no longer needed, to simplify this a bit.

OkHttp 
AuthTokenRefreshingInterceptor
AccessTokenProvider
AuthTokenService
AuthEventBroadcaster
AuthEventListener(s, list)
HAnalytics
OkHttp

Then the final fix was that `HAnalyticsExperimentManagerImpl` needed to be the storage for the experiments, which had to be cleared when logging in and out for example, but that itself also needed access back to `AndroidHAnalyticsService` which needed `OkHttpClient` and so on, so this was the final issue. Extracting `HAnalyticsExperimentStorageImpl` outside, so that the storage part is standalone and can be created alone without any other dependencies was the fix. So now the things that need to clear the Experiments list can depend on that directly, instead of the entire HAnalytics.
Anyway, this is a bit hard to explain, but deciphering the circular dependency was quite a difficult thing, which I know now is solved at least :D

### CODE REVIEW

**Review by hugokallstrom — APPROVED** [review]
Really good 👌

File: app/feature/feature-login/src/main/kotlin/com/hedvig/android/feature/login/swedishlogin/SwedishLoginViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.hedvig.android.feature.login.swedishlogin
+
+import com.hedvig.android.auth.AuthTokenService
+import com.hedvig.android.molecule.android.MoleculeViewModel
+import com.hedvig.authlib.AuthRepository
+import kotlinx.coroutines.flow.SharingStarted
+
+internal class SwedishLoginViewModel(
+  private val authTokenService: AuthTokenService,
```
[the remark this answers is not in the document]
Comment (StylianosGakis) [reply app/feature/feature-login/src/main/kotlin/com/hedvig/android/feature/login/swedishlogin/SwedishLoginViewModel.kt:9]: Addressed in https://github.com/HedvigInsurance/android/pull/1788

File: app/feature/feature-login/src/main/kotlin/com/hedvig/android/feature/login/swedishlogin/SwedishLoginPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      autoStartToken = SwedishLoginUiState.HandlingBankId.AutoStartToken(bankIdPropertiesValue.autoStartToken),
+      loginStatusResult = loginStatusResult,
+      allowOpeningBankId = allowOpeningBankId,
+      navigateToLoginScreen = navigateToLoginScreen,
+    )
+  }
+}
+
+internal sealed interface SwedishLoginEvent {
+  object Retry : SwedishLoginEvent
```
[the remark this answers is not in the document]
Comment (StylianosGakis) [reply app/feature/feature-login/src/main/kotlin/com/hedvig/android/feature/login/swedishlogin/SwedishLoginPresenter.kt:129]: Addressed in https://github.com/HedvigInsurance/android/pull/1788

File: app/apollo/apollo-octopus-public/src/main/graphql/com/hedvig/android/apollo/octopus/graphql/insurance/ProductVariantFragment.graphql [hunk 3]
```diff
@@ -2,12 +2,6 @@ fragment ProductVariantFragment on ProductVariant {
   displayName
   typeOfContract
   partner
-  product {
```
[the remark this answers is not in the document]
Comment (StylianosGakis) [reply app/apollo/apollo-octopus-public/src/main/graphql/com/hedvig/android/apollo/octopus/graphql/insurance/ProductVariantFragment.graphql:5]: Perfect!

File: app/feature/feature-login/src/main/kotlin/com/hedvig/android/feature/login/swedishlogin/SwedishLoginPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+          }
+      }
+    }.collectAsState(initial = null).value
+
+    LaunchedEffect(retryIndex) {
+      val result = authRepository.startLoginAttempt(LoginMethod.SE_BANKID, Market.SE.name)
+      when (result) {
+        is AuthAttemptResult.BankIdProperties -> bankIdProperties = result
+        is AuthAttemptResult.Error -> {
+          logcat(LogPriority.ERROR) { "Got Error when signing in with BankId: ${result.message}" }
```
[the remark this answers is not in the document]
Comment (StylianosGakis) [reply app/feature/feature-login/src/main/kotlin/com/hedvig/android/feature/login/swedishlogin/SwedishLoginPresenter.kt:76]: Yeap, through https://github.com/HedvigInsurance/android/blob/develop/app/datadog/src/main/kotlin/com/hedvig/android/datadog/DatadogLoggingTree.kt

**Pull request merged**